### ISSUE TITLE
Architecture x86_64-apple-ios-simulator

### ISSUE THREAD

**Issue by acrabb** [omise/omise-ios#119]
Running our project that used to work in Xcode 11 gives the following error for Omise:
```
Could not find module 'OmiseSDK' for target 'x86_64-apple-ios-simulator'; found: arm64, armv7-apple-ios, arm64-apple-ios, arm, armv7
```

Does x86_64-apple-ios-simulator exist in any specific build version?
I've ran 3.8.0, 3.7.1, and 3.3.0

**Comment by truongvnse03130** [comment]
Any updates for it?

**Comment by dsasmita** [comment]
any update on this? having same issues 

```
Swift Compiler Error (Xcode): Could not find module 'OmiseSDK' for target 'x86_64-apple-ios-simulator'; found: arm64-apple-ios-simulator
```

**Comment by punyapat-s** [comment]
Any update? Apple suggestion https://developer.apple.com/documentation/technotes/tn3117-resolving-build-errors-for-apple-silicon

**Comment by vault087** [comment]
Hello @punyapat-s @dsasmita @acrabb, we were inspecting this issue but unfortunately couldn't reproduce it in various environments. Could you please provide us more information about which package manager do you use, Xcode and MacOS versions, is it Swift of Objective-C project, and any other related information that could help us to reproduce this issue.

**Comment by tpeodndyy** [comment]
@vault087 
This issue is happening after installing Xcode 14.3. I was working fine on Xcode 14.2 and 14.1, believed to cause by the simulator Rosetta update. 

my project is Swift project. I tried adding several workaround but it's not working. Just to point out, it's working fine with Real Device and for Appstore build since it's not for x86_64-apple-ios-simulator.

**Comment by ej-redplanet** [comment]
In my opinion, OmiseSDK must provide a file that will support x86_64 in OmiseSDK.swiftmodule folder since it only has arm64.

**Comment by vault087** [comment]
It seems to me you are using Carthage and we are no longer support it. Could you please switch to use Swift Package Manager for OmiseSDK and let me know if you still having any issues.

@tpeodndyy @ej-redplanet @punyapat-s

**Comment by aashishgurung** [comment]
We’re going to close this since there hasn’t been any activity for a while. If this issue is still relevant, please feel free to reopen it with more details and we’ll take another look.

**Comment by AgungLaksana** [comment]
> It seems to me you are using Carthage and we are no longer support it. Could you please switch to use Swift Package Manager for OmiseSDK and let me know if you still having any issues.
> 
> @tpeodndyy @ej-redplanet @punyapat-s

I also have this error:
Swift Compiler Error (Xcode): Could not find module 'OmiseSDK' for target 'x86_64-apple-ios-simulator'; found: arm64-apple-ios-simulator

I install the Omise ioS SDK using Swift Package Manager, not Carthage. could you please help?

I am using Macbook Pro 2021 (M1)

I install the SDK like this: https://i.stack.imgur.com/jecvW.jpg

**Comment by vault087** [comment]
@AgungLaksana please set Dependency Rule to be "Exact Version" or "Up to Next Major Version", the latest version at the moment it `4.27.1`

<img width="459" alt="image" src="https://github.com/omise/omise-ios/assets/17623015/a3aff6c8-f8eb-4766-8439-5ac0fdf12803">

**Comment by AgungLaksana** [comment]
@vault087 

Thanks for the response.

I have tried to use `4.27.1` like this https://i.stack.imgur.com/WrtOs.jpg
but I still have the same error message

> 
> Swift Compiler Error (Xcode): Could not find module 'OmiseSDK' for target 'x86_64-apple-ios-simulator'; found: arm64-apple-ios-simulator, at: /Users/xxx/Documents/mobile_app_project/build/ios/Debug-development-iphonesimulator/OmiseSDK.swiftmodule
> /Users/xxxo/Documents/blueraycargo_mobile_app_project/ios/Runner/services/Omise.swift:8:7

**Comment by vault087** [comment]
@AgungLaksana based on this [thread](https://forums.developer.apple.com/forums/thread/123614) could you check that `Embed` settings for OmiseSDK is set to `Embed & Sign`:

<img width="903" alt="image" src="https://github.com/omise/omise-ios/assets/17623015/798a0625-fe84-43f1-93e1-4c42f86d3ed2">

**Comment by AgungLaksana** [comment]
@vault087 

I think I have set the embed as well, but unfortunately I still get the same error

<img width="1440" alt="Screenshot 2024-03-06 at 16 09 06" src="https://github.com/omise/omise-ios/assets/30435913/c4fd205d-0807-4954-9c03-0c3c4cf03d48">

**Comment by vault087** [comment]
@AgungLaksana It seems you're using Flutter, we will investigate this issue and get back to you!

**Comment by AgungLaksana** [comment]
> @AgungLaksana It seems you're using Flutter, we will investigate this issue and get back to you!

yup thats right, I use Flutter, but I install Omise SDK natively using Swift and integrate it back to Flutter using [Platform Channel](https://docs.flutter.dev/platform-integration/platform-channels?tab=type-mappings-kotlin-tab)

the SDK will have no issue if I run it on real device, the problem only occured on iOS simulator. I use

- Xcode 15.0.1
- MacBook Pro 13-inch, M1, 2020
- Flutter 3.16.4

feel free to message me if you need more info :)

**Comment by Hiteshboricha-Pomelo** [comment]
Hello team, I removed Carthage and  added OmiseSDK using Swift Package Manager but still I am getting the same issue. 


This issue is happening after installing Xcode 15.3. I was working fine on Xcode 14.2 and 14.1, believed to cause by the simulator Rosetta update.

my project is Swift project. I tried adding several workaround but it's not working. Just to point out, it's working fine with Real Device and for Appstore build since it's not for x86_64-apple-ios-simulator.

![Screenshot 2024-04-29 at 7 06 47 PM](https://github.com/omise/omise-ios/assets/152582994/4ab46584-95ce-426e-86d4-0f683de05df8)


![Screenshot 2024-04-30 at 3 02 23 PM (1)](https://github.com/omise/omise-ios/assets/152582994/d82f63c4-fdcb-4b9f-a746-124cfbc5ef80)

![Screenshot 2024-04-30 at 3 01 15 PM](https://github.com/omise/omise-ios/assets/152582994/063fbf96-6cfe-41d7-bdc4-8756ae79bfa7)
